crates/telemetry-subscribers/src/lib.rs

```diff
@@ -155,23 +164,24 @@ impl TracingHandle {
                     error!("failed to reset trace filter: {}", e);
                 }
             });
-            res
+            Ok(())
         } else {
             info!("tracing not enabled, ignoring update");
-            Ok(())
+            Err(TelemetryError::TracingDisabled)
         }
     }
 
     pub fn clear_file_output(&self) {
         self.file_output.clear_path();
     }
 
-    pub fn reset_trace(&self) {
+    pub fn reset_trace(&self) -> Result<(), TelemetryError> {
         if let Some(trace) = &self.trace {
             let trace_filter_env = env::var("TRACE_FILTER").unwrap_or_else(|_| "off".to_string());
-            if let Err(e) = trace.update(trace_filter_env) {
-                error!("failed to reset trace filter: {}", e);
-            }
+            trace.update(trace_filter_env).map_err(|e| e.into())
+        } else {
+            info!("tracing not enabled, ignoring reset");
```

Comment:
```suggestion
```
I guess it's fine without the info message, you will see it in the request itself.